src/encoder.rs

```diff
@@ -1723,14 +1723,24 @@ fn encode_partition_bottomup(seq: &Sequence, fi: &FrameInvariants, fs: &mut Fram
             rd_cost = (w.tell_frac() - tell) as f64 * get_lambda(fi, seq.bit_depth)/ ((1 << OD_BITRES) as f64);
         }
 
-        rd_cost += encode_partition_bottomup(seq, fi, fs, cw, w_pre_cdef, w_post_cdef, subsize,
-                                             bo);
-        rd_cost += encode_partition_bottomup(seq, fi, fs, cw, w_pre_cdef, w_post_cdef, subsize,
-                                             &BlockOffset { x: bo.x + hbs as usize, y: bo.y });
-        rd_cost += encode_partition_bottomup(seq, fi, fs, cw, w_pre_cdef, w_post_cdef, subsize,
-                                             &BlockOffset { x: bo.x, y: bo.y + hbs as usize });
-        rd_cost += encode_partition_bottomup(seq, fi, fs, cw, w_pre_cdef, w_post_cdef, subsize,
-                                             &BlockOffset { x: bo.x + hbs as usize, y: bo.y + hbs as usize });
+        {
+            let mut split = |offset: &BlockOffset| {
+                rd_cost += encode_partition_bottomup(
+                    seq,
+                    fi,
+                    fs,
+                    cw,
+                    w_pre_cdef,
+                    w_post_cdef,
+                    subsize,
+                    offset
+                );
+            };
+            split(bo);
+            split(&BlockOffset { x: bo.x + hbs as usize, y: bo.y });
+            split(&BlockOffset { x: bo.x, y: bo.y + hbs as usize });
+            split(&BlockOffset { x: bo.x + hbs as usize, y: bo.y + hbs as usize });
+        }
 
```

Comment:
you may even do something like:
``` rust
let partitions = [bo, 
  &BlockOffset { x: bo.x + hbs as usize, y: bo.y }, 
  &BlockOffset { x: bo.x, y: bo.y + hbs as usize }, 
  &BlockOffset { x: bo.x + hbs as usize, y: bo.y + hbs as usize }
];
```
and then

``` rust
rd_cost += partitions.iter().for_each(|&offset| encode_partition_bottomup(
                    seq,
                    fi,
                    fs,
                    cw,
                    w_pre_cdef,
                    w_post_cdef,
                    subsize,
                    offset
                )
).sum();

and something like that for the rest,

Reply:
Yes, this was exactly what I was talking about, and I personally prefer it because it keeps the original function name which I search by it often to trace where they are used.